### ISSUE TITLE
Vulkan BadDrawable

### ISSUE THREAD

**Issue by rares45** [godotengine/godot#43714]
**Godot version:** v4.0.dev from master -https://github.com/godotengine/godot/commit/6a683f80169850d1138c38f7182bdcadf4dafd10

**OS/device including version:** 
Linux Mint 20 Cinnamon
GeForce GTX 1050/PCIe/SSE2

**Issue description:**
When I try to open Godot it fails. The output:

```
Godot Engine v4.0.dev.custom_build - https://godotengine.org
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  149 ()
  Minor opcode of failed request:  4
  Resource id in failed request:  0x5600006
  Serial number of failed request:  73
  Current serial number in output stream:  79
```
I know there are more issues like this, but they are solved by restarting the os, using Wayland, or ensuring that you run on the GPU, but I did all of this, excepting using Wayland because I can't. If you can not help me please tell me how to debug the app, because I am willing to contribute.

**Steps to reproduce:**
1. Compile Godot from master using `scons -j 8 platform=linuxbsd tools=yes target=release_debug`
2. Run it.

**Comment by Calinou** [comment]
Does it work if you leave out `target=release_debug` (which uses `target=debug` instead)?

**Comment by rares45** [comment]
> Does it work if you leave out `target=release_debug` (which makes it use `target=debug` instead)?

No, it's the same error.

**Comment by bruvzg** [comment]
> Major opcode of failed request:  149

This is an error from one of X extensions, opcodes are dynamically assigned.
Please try running Godot again, and provide new opcodes from the error and output of `xdpyinfo -queryExt | grep opcode` command executed in the same X session, to see which one is responsible?

**Comment by rares45** [comment]
The output is 
`DRI3  (opcode: 149)`

**Comment by bruvzg** [comment]
It's request from the sync mechanism between X server and kernel, not sure it's gonna help much, seems like a driver issue.

Full failed request info:
```
┌───
    DRI3FenceFromFD
	drawable: DRAWABLE
	fence: FENCE
	initially-triggered: BOOL
	fd: FD
└───
	Errors: IDchoice, Drawable

	Creates a Sync extension Fence that provides the regular Sync
	extension semantics along with a file descriptor that provides
	a device-specific mechanism to manipulate the fence directly.
	Details about the mechanism used with this file descriptor are
	outside the scope of the DRI3 extension.
```

Are #38225 and #42348 the same issue?

**Comment by rares45** [comment]
I tried reinstalling the drivers, but no luck. I use the Nvidia proprietary drivers.

 #38225 and #42348 do look the same.

**Comment by danilw** [comment]
this problem exists on Linux in Nvidia for all Vulkan apps (all apps that draw more than 6 triangles)

for some reason after closing Vulkan app it can not be launched again, it reproduced even on Khronos samples(their HDR demos and compute shader demo can not be launched again before reboot)

I had a similar problem there https://github.com/DiligentGraphics/DiligentSamples/issues/25

(this "bug" randomly fixes by launching OpenGL app, for me most of times X11 error on Vulkan app launch is gone after 2-3 mins using other OpenGL app, and Vulkan start working again)

I do not have this problem with Godot 4.0 on Linux (Nvidia 750, latest drivers), but it "startup" after first time takes minutes...

**Comment by rares45** [comment]
I tried running Godot 4.0 using my CPU(i5-8250U) and it worked. So that confirms it's an Nvidia driver issue. But somehow CS:GO, which is using Vulkan works perfectly.

Should I also open an issue on Nvidia's page?

**Comment by danilw** [comment]
> But somehow CS:GO, which is using Vulkan works perfectly.
games under DXVK work correctly when they "simple" like CS:GO that do not use complex graphic
as an example - game Risk of Rain 2(from steam in their Proton with DXVK) can not be launched second time after closing, only after ~1min waiting/using OpenGL app, it spam X11 error

> Should I also open an issue on Nvidia's page?
you can try... I did not report it to Nvidia (using their forum) because this bug very complex, and ... no one will debug "your app"(Godot), even if you recreate this bug in Khronos samples, that "longer" than a single triangle code... no one gonna waste hours to debug it

reporting bugs in Vulkan is problematic due to the size of code to analyze/debug

**Comment by Calinou** [comment]
> But somehow CS:GO, which is using Vulkan works perfectly.

CS:GO uses OpenGL (translated from Direct3D using Valve's ToGL library). Like all games translated using ToGL, it has notorious performance issues on Linux.

Running it via WINE/Proton used to be an option (which effectively let you use D9VK to translate Direct3D 9 to Vulkan), but now that VAC no longer works in WINE, it's not really an option anymore.

**Comment by rares45** [comment]
> CS:GO uses OpenGL (translated from Direct3D using Valve's [ToGL](https://github.com/ValveSoftware/ToGL) library).

Didn't know that.

> and ... no one will debug "your app"

well said

After all, what do I do? I wait for Nvidia Drivers update or this bug can be fixed from Godot sides? 
I want to contribute to Godot, but I can't without starting the app. I could try myself to fix this, but I don't know where to start.

**Comment by danilw** [comment]
> CS:GO uses OpenGL 

it can be launched from Windows version(in Linux steam), that will use Proton with DXVK as I know, I thought it launched this way

**Comment by rares45** [comment]
> use Windows

not an option

> or get new 2xxx/3xxx video card, or get AMD videocard

I would love to, but I am on a laptop.

> they will drop Vulkan support

Most probably they would do that (because Nvidia is trash, at least on Linux), but the 10' series is not old yet. There are still a lot of new laptops and builds that come with 10' series cards.

> amdgpu driver opensource with active development and live bugfixes for all supported videocards

You reminded me of the fact that there is an open-source driver for Nvidia too, and I should check if there is the same problem.

**Comment by Calinou** [comment]
> You reminded me of the fact that there is an open-source driver for Nvidia too, and I should check if there is the same problem.

The Nouveau driver does not support reclocking on Maxwell 2 and newer GPUs, which means you will get very low performance (5× slower than the proprietary driver).

**Comment by qarmin** [comment]
I have similar error but when creating 2D Node with Godot with GLES 2 (https://github.com/godotengine/godot/pull/44399) on Laptop with Intel HD 3000 and Mesa 21.0.0-devel (git-9d59c84e31),
```
WARNING: XCreateIC couldn't create wd.xic
     at: _create_window (platform/linuxbsd/display_server_x11.cpp:3808)
rendering_driver GLES2
window_create window id 1
DisplayServer::_create_window 1 want rect: 350, 100, 900, 700 got rect 350, 100, 900, 700
DisplayServer::_window_changed: 1 rect: 350, 124, 900, 700
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  154 (DRI2)
  Minor opcode of failed request:  7 (DRI2GetBuffersWithFormat	)
  Resource id in failed request:  0x4800051
  Serial number of failed request:  4475
  Current serial number in output stream:  4475
```

**Comment by lawnjelly** [comment]
> I have similar error but when creating 2D Node with Godot with GLES 2 (#44399) on Laptop with Intel HD 3000 and Mesa 21.0.0-devel (git-9d59c84e31),

Just seen this - I'm not sure this is related to the original issue. It is highly likely to be me doing something wrong in the X code, I've only tested GLES2 on my own machine so far, and there's been a lot of guesswork involved especially with multi-window. It will have to go through some testing / bug fixing. Sorry been a little delay, I've been temporarily working on BVH as it will help reduz with the rendering server rewrite.

**Comment by danilw** [comment]
tested today latest Godot 4 Vulkan build (just build), with latest Nvidia Linux driver 455.45.01
**can confirm - Godot 4 absolutely unusable** on Linux

Godot 4 freeze on any "text editing", also on opening any menu(popup), Godot 4 even freeze in startup window on creating "new project"(because it has editable text line)

in Godot case I not sure if it Nvidia driver problem or is Godot problem
because **launching Godot with Vulkan validation layer result lots of errors**
on pressing "New Project" I see this output(and Godot 4 freeze same as without validation layers)

full log [pastebin link ](https://pastebin.com/de6JzSXp)

**Comment by Zireael07** [comment]
NVidia driver 455 has problems that were already reported in another issue, and at least two recent dupes - and since rolling back fixes them, I think they are on NVidia side.

**Comment by Calinou** [comment]
@danilw See https://github.com/godotengine/godot/issues/41614.

**Comment by rares45** [comment]
Update: It still does not start. But this time is a new output:

```
Using Vulkan Device #0: INTEL - Intel(R) UHD Graphics 620 (KBL GT2)
- Vulkan multiview supported:
  max views: 16
  max instances: 268435455
- Vulkan subgroup:
  size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, FEATURE_CLUSTERED, FEATURE_QUAD
  quad operations in all stages
X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  149 ()
  Minor opcode of failed request:  4
  Resource id in failed request:  0x4a00008
  Serial number of failed request:  110
  Current serial number in output stream:  116
```
The interesting part is the selected device, because I have my profile set on NVIDIA (performance mode), yet the selected device is the integrated graphics. Also if I set the profile on Intel, Godot starts without a problem so the integrated graphics support Vulkan.

Godot version: 4.0 from master - https://github.com/godotengine/godot/commit/9539e4e39e489e750c7ec99d84b93dce283fe721 \
This time I'm on Ubuntu instead of Linux Mint. \
Drivers: nvidia-driver-460

Also, I don't think this is a driver issue(or at least entirely) because Vulkan demos and Unreal Engine 5 runs perfectly.

Just wanted to let you know. Sorry for my English.

**Comment by Calinou** [comment]
@rares45 Can you test [this pull request](https://github.com/godotengine/godot/pull/46070) by compiling it locally?

**Comment by rares45** [comment]
Yes, it works. Now it starts and runs. This seems to solve my issue.

**Comment by rares45** [comment]
Now when trying to create a new sky the Editor freezes. That didn't happen when running with Intel. Is this a known bug or a result of #46070?

**Comment by Calinou** [comment]
> Now when trying to create a new sky the Editor freezes. That didn't happen when running with Intel. Is this a known bug or a result of #46070?

Did you create a ProceduralSky or PhysicalSky resource? I'd recommend opening a new issue for that.

**Comment by rares45** [comment]
It freezes when trying to create the Sky component, right before pressing `New Sky`. It happens every time. I will start a new issue after I find out more
